### ISSUE TITLE
Catch 'metadata not found' error 

### ISSUE THREAD

**Issue by culdav** [dimitrov-d/angular-url-preview#1]
Hi @dimitrov-d,

I'm trying to display a preview of some URLs in my project, but for some of them it seems that the proxy call is failing with status code 403. Would it be possible to provide an event handler to handle this case?  I would like to hide the preview if the metadata for the provided URL can't be obtained. 

Here's a link for which the proxy call is failing: https://cors-proxy.xyz/fetch/https://www.baeldung.com/spring-security-oauth

Thanks in advance,
Vlad

**Comment by dimitrov-d** [comment]
Hi @culdav, thanks for opening an issue.

This is actually intentional which you can see in the code, it throws an error to indicate that the URL should not be used if metadata cannot be obtained. That URL seems to be accessible behind a firewall so that's why the scraper does not work.

Could you explain why you would want that error to be caught, or in fact not thrown at all? Is it a scenario where you have dynamic URL preview components and you cannot predict if the URL is going to work or not?

**Comment by culdav** [comment]
Yes, it's a feature in which I have some comments and I would like that when I paste a URL, to see a preview of that URL. I could make use of the "customImageSrc" property, but when I set it, it seems that it will always display that custom image even if the URL has the metadata tags provided.
